chore: merger task

nitzanmorr opened the pull request.

nitzanmorr committed:
before pr

razbroc commented:
not enough commits + not detailed commits descriptions
make sure you follow best practice syntax (chroe/fix...: {commit description})
we would want to commit to every big change that we do.
 (I have also looked at the main)

nitzanmorr committed:
Fix Program.cs issues

Made the check for null paths more efficiant and readable, made better naming, better error handling and moved the connection closing to the main method

nitzanmorr committed:
Fix: Utils.cs issues

Applied dependancy injection in SetNewExtent query, made better namings and better error handling

razbroc commented on Program.cs:
```diff
@@ -1,13 +1,21 @@
-if (Environment.GetEnvironmentVariable("GPKG_A") == null || Environment.GetEnvironmentVariable("GPKG_B") == null)
+string firstGpkgPath = Environment.GetEnvironmentVariable("GPKG_A") ?? throw new NullReferenceException("GPKG_A cannot be null");
+string secondGpkgPath = Environment.GetEnvironmentVariable("GPKG_B") ?? throw new NullReferenceException("GPKG_B cannot be null");
+SQLiteConnection sqlite_connA = EstablishConnection(firstGpkgPath);
```
`sqlite_connA` is not a best practice for variables names.
read about [camel case](https://www.techtarget.com/whatis/definition/CamelCase)

razbroc commented on Utils.cs:
```diff
@@ -1,22 +1,22 @@
         try
         {
             sqlite_conn.Open();
+            return sqlite_conn;
```
variable name?

nitzanmorr committed:
Fix: namings by conventions

razbroc commented on Utils.cs:
```diff
@@ -1,80 +1,77 @@
+        $@"ATTACH DATABASE '{connB.FileName}' AS {secondGpkgAlias};
+        INSERT OR IGNORE INTO {tableNameA} (zoom_level, tile_column, tile_row, tile_data)
+        SELECT zoom_level, tile_column, tile_row, tile_data
         FROM gpkgB.{tableNameB};";
```
you declared a variable `secondGpkgAlias` and then used `gpkgB`

razbroc commented on Utils.cs:
```diff
@@ -1,80 +1,77 @@
+        catch
         {
-            throw new Exception(e.Message);
+            throw;
```
What is this?

razbroc commented on Consts.cs:
```diff
@@ -0,0 +1,4 @@
+class Consts
```
I think that declaring a Consts class for one variable is a bit overkill.
You use this variable once, you can declare it in function scope and that should be enough.

razbroc commented on Utils.cs:
```diff
@@ -1,80 +1,77 @@
     }
 
-    private static void SetNewExtent(SQLiteConnection connA)
+    private static void SetNewExtent(SQLiteConnection connA, string gpkgBAlias)
```
the meaning of the variable is not a `gpkgBAlias` it's just the `gpkgB`
